podpac/core/data/rasterio_source.py

```diff
@@ -119,6 +135,14 @@ def get_data(self, coordinates, coordinates_index):
     # additional methods and properties
     # -------------------------------------------------------------------------
 
+    @property
+    def tags(self):
+        return self.dataset.tags()
+
+    @property
+    def subdatasets(self):
+        return self.dataset.subdatasets
```

Comment:
We could potentially make the subdatasets as different `outputs` and then open multiple datasets to composite the results. That would take another branch in the `get_data` function but would results in a much nicer user experience.

Reply:
Yeah. A dataset can already have multiple outputs, at least if you open it with the xarray `Dataset` or `H5PY` nodes. Could you end up with a situation with several subdatasets that each have multiple outputs, but no way to have "nested" outputs?

Reply:
that makes me want to scream -- I don't know. I hope the answer is NO but I suspect it's a resounding Yes.

Reply:
Well, the simple answer then is not to make the subdatasets different outputs, basically require one node per subdataset. Hopefully the file can be open concurrently for multiple read-only file-pointers.

Reply:
Based on the research I've done, there should not be any multi-band subdatasets. 

I think I'll end up making the subdatasets multiple outputs -- make it really easy for users. I've now heard from two people that they gave up on PODPAC because they couldn't read the HDF files and they didn't know why. So, it should just work without them having to know about the intricacies of subdatasets. 

Also, the s3 case is not a real use-case since reading and hdf file from S3 seems to involve reading the whole file anyway. In other words, it's very slow. 

A related questions: right now we read band 1 by default. Should we just go a head and read all bands by default? What do you think.

Reply:
Okay. Yeah, I guess we should read all of the bands by default. The `FileKeysMixin` that is used by `CSV`, `Zarr`, `Dataset`, and `H5PY` loads all available `data_keys` by default as separate `outputs`, unless there is only one, in which case `outputs` is None and the node is a standard single-output node.

```python
    @tl.default("data_key")
    def _default_data_key(self):
        if len(self.available_data_keys) == 1:
            return self.available_data_keys[0]
        else:
            return self.available_data_keys

    @tl.default("outputs")
    def _default_outputs(self):
        if not isinstance(self.data_key, list):
            return None
        else:
            return self.data_key
```

`Rasterio` should be basically the same, something like allowing `band` to be either a list or a single value and using this:

```python
    @tl.default("band")
    def _default_band(self):
        if len(self.band_keys == 1:
            return self.band_keys[0]
        else:
            return self.band_keys

    @tl.default("outputs")
    def _default_outputs(self):
        if not isinstance(self.band, list):
            return None
        else:
            return self.band
```

Reply:
Do you want to make an issue for this, or fix it now before merging this issue? Do you have time, or shall I?

Reply:
I think fixing this inside this PR is fine. And don't worry about timing -- this is complex enough that I think we should do it for the next release and not rush something.